scylla/src/transport/load_balancing.rs

```diff
@@ -210,26 +206,26 @@ impl LoadBalancingPolicy for TokenAwarePolicy {
 
                 match strategy {
                     Strategy::SimpleStrategy { replication_factor } => {
-                        let mut replicas =
+                        let replicas =
                             Self::simple_strategy_replicas(cluster, &token, *replication_factor);
-                        self.child_policy.apply_child_policy(&mut replicas)
+                        self.child_policy.apply_child_policy(replicas)
                     }
                     Strategy::NetworkTopologyStrategy {
                         datacenter_repfactors,
                     } => {
-                        let mut replicas = Self::network_topology_strategy_replicas(
+                        let replicas = Self::network_topology_strategy_replicas(
                             cluster,
                             &token,
                             datacenter_repfactors,
                         );
-                        self.child_policy.apply_child_policy(&mut replicas)
+                        self.child_policy.apply_child_policy(replicas)
                     }
                     _ => {
                         // default to simple strategy with replication factor = 1
                         let replication_factor = 1;
-                        let mut replica =
+                        let replica =
                             Self::simple_strategy_replicas(cluster, &token, replication_factor);
-                        self.child_policy.apply_child_policy(&mut replica)
+                        self.child_policy.apply_child_policy(replica)
                     }
                 }
```

Comment:
I think this whole `match` can be simplified:

```rust
let replicas = match strategy {
    Strategy::SimpleStrategy { replication_factor } => {
            Self::simple_strategy_replicas(cluster, &token, *replication_factor)
    }
    Strategy::NetworkTopologyStrategy {
        datacenter_repfactors,
    } => {
        Self::network_topology_strategy_replicas(
            cluster,
            &token,
            datacenter_repfactors,
        )
    }
    _ => {
        // default to simple strategy with replication factor = 1
        let replication_factor = 1;
        Self::simple_strategy_replicas(cluster, &token, replication_factor)
    }
};

self.child_policy.apply_child_policy(replicas)
```

Reply:
Done